### PR TITLE
Added tar.gz for Linux release

### LINKED ISSUE #8052: Change to tar for Linux release?
Would it be possible to package future Linux releases using tarball instead of zip as is currently the case? Tar is built into pretty much every *nix system, and its respect and support is much better. Would greatly appreciate it.

### PR DESCRIPTION
**Pull request by colorizer**
This changes the default packaging format to tar.gz for linux releases.

Fixes #8052 

## PR checklist

* [x] Code
	* [ ] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: editor/scripts/bundle.py [hunk 1]
```diff
@@ -113,19 +114,28 @@ def download(url):
         log('Downloading %s failed' % (url))
     return path
 
-def ziptree(path, outfile, directory = None):
+def archivetree(path, outfile, directory = None, fmt= 'zip'):
+    # Generate archvie of fmt zip or tar.gz
```
Comment (britzl) [inline editor/scripts/bundle.py:118]: archvie -> archive

Comment (colorizer) [reply editor/scripts/bundle.py]: corrected.

File: editor/scripts/bundle.py [hunk 2]
```diff
@@ -113,19 +114,28 @@ def download(url):
...
     # Directory is similar to -C in tar
+    if fmt=='zip':
```
[the remark this answers is not in the document]
Comment (colorizer) [reply editor/scripts/bundle.py]: updated.

**Comment by colorizer** [comment]
Please note that I haven't updated the `download_editor2` function for the `tar.gz` support. https://github.com/defold/defold/blob/1d116d7331e1b62909da48369e410e09344fafad/scripts/build.py#L1362-L1370

Should the above be updated as well?

**Comment by vlaaad** [comment]
@colorizer let's start by aligning our goals before diving into the PR discussion, please take a moment to consider https://github.com/defold/defold/issues/8052#issuecomment-1750215012.

**Comment by colorizer** [comment]
Reverted previous changes and added tar.gz as additional bundle.

**Comment by vlaaad** [comment]
@colorizer hey, I must have missed this PR somehow, it's ready for review, right?

**Comment by colorizer** [comment]
@vlaaad yes.

**Review by vlaaad — APPROVED** [review]
LGTM, but we want the PR to be based off `editor-dev` and targeting `editor-dev` branch. See https://github.com/defold/defold/pull/8133

**Comment by vlaaad** [comment]
@colorizer thank you for your contribution ❤️
Since this PR does not touch engine, we can make it available to our users sooner by rebasing the changes to `editor-dev` branch and targeting `editor-dev`. I cherry-picked your commits without changes here https://github.com/defold/defold/pull/8133

**Pull request closed without merging**